src/xrpld/app/misc/CredentialHelpers.cpp

```diff
@@ -225,6 +226,50 @@ makeSorted(STArray const& in)
     return out;
 }
 
+NotTEC
+checkArray(STArray const& credentials, unsigned maxSize, beast::Journal j)
+{
+    if (credentials.empty() || (credentials.size() > maxSize))
+    {
+        JLOG(j.trace()) << "Malformed transaction: "
+                           "Invalid credentials size: "
+                        << credentials.size();
+        return temMALFORMED;
+    }
```

Comment:
This would be slightly easier to debug if it used `temARRAY_EMPTY` and `temARRAY_TOO_LARGE`

Reply:
added